apps/linkedin-to-notion/src/background/index.ts

```diff
@@ -4,30 +4,43 @@ export {};
 
 // You can test the regex here: https://regex101.com/r/RJgYar/2
 // Don't forget to update the comment when you update the regex on regex101.com
-export const linkedInProfileURLRegex = /linkedin\.com\/in\/[^/]+\/#?/;
+export const linkedInProfileURLRegex = /linkedin\.com\/in\/[^/]+\/#?$/;
 export const linkedInURLRegex = /linkedin\.com/;
 
 chrome.tabs.onUpdated.addListener(async (tabId, { status }, tab): Promise<void> => {
   if (status !== 'complete') {
     return;
   }
 
-  const isOnLinkedIn = tab?.url?.match(linkedInURLRegex);
-  const isOnLinkedInProfile = tab?.url?.match(linkedInProfileURLRegex);
+  if (!tab?.url) {
+    return;
+  }
+
+  const isOnLinkedIn = linkedInURLRegex.test(tab.url);
+  const isOnLinkedInProfile = linkedInProfileURLRegex.test(tab.url);
 
   // We close the extension when we leave LinkedIn
   if (!isOnLinkedIn) {
     try {
-      console.log(tab?.url);
+      console.log(tab.url);
       await chrome.tabs.sendMessage(tabId, 'closeSidePanels');
     } catch (e) {
       console.log('Error closing side panels', e);
     }
   }
 
+  if (!isOnLinkedInProfile) {
+    try {
+      await chrome.tabs.sendMessage(tabId, 'closeSidePanels');
+    } catch (e) {
+      console.log('Error updating side panels', e);
+    }
+  }
+
   // We update the side panel's content when we are on a LinkedIn profile
   if (isOnLinkedInProfile) {
     try {
+      await chrome.tabs.sendMessage(tabId, 'openLinkedInNotionSidePanel');
```

Comment:
I think you should delete this line. cf. my upcoming PR, we shouldn't rely on content script messaging to open the side panel imo. There's already a matching condition for that in the content script

Reply:
Well, after working on this I tend to disagree on the "we shouldn't rely on content script messaging". Those work perfectly and are super handy. We were just using it wrong